frontend/components/ActivityItem/ActivityItem.tsx

```diff
@@ -0,0 +1,177 @@
+import React from "react";
+import ReactTooltip from "react-tooltip";
+import classnames from "classnames";
+
+import { IActivity, IActivityDetails } from "interfaces/activity";
+import {
+  addGravatarUrlToResource,
+  internationalTimeFormat,
+} from "utilities/helpers";
+import { DEFAULT_GRAVATAR_LINK } from "utilities/constants";
+
+import Avatar from "components/Avatar";
+
+import { COLORS } from "styles/var/colors";
+import { dateAgo } from "utilities/date_format";
+import Button from "components/buttons/Button";
+import Icon from "components/Icon";
+import { noop } from "lodash";
+
+const baseClass = "activity-item";
+
+export interface IShowActivityDetailsData {
+  type: string;
+  details?: IActivityDetails;
+}
+
+/**
+ * A handler that will show the details of an activity. This is used to pass
+ * the details of an activity to the parent component to show the details of
+ * the activity.
+ */
+export type ShowActivityDetailsHandler = ({
+  type,
+  details,
+}: IShowActivityDetailsData) => void;
+
+interface IActivityItemProps {
+  activity: IActivity;
+  children: React.ReactNode;
+  /**
+   * Set this to `true` when rendering only this activity by itself. This will
+   * change the styles for the activity item for solo rendering.
+   * @default false */
+  soloActivity?: boolean;
```

Comment:
can we put `is` on this, so `isSoloActivity`?

Reply:
yep updated